v1/ast/parser.go

```diff
@@ -1667,12 +1671,19 @@ func (p *Parser) parseNumber() *Term {
 		}
 	}
 
-	// Check for multiple leading 0's, parsed by math/big.Float.Parse as decimal 0:
-	// https://golang.org/pkg/math/big/#Float.Parse
-	if ((len(prefix) != 0 && prefix[0] == '-') || len(prefix) == 0) &&
-		len(p.s.lit) > 1 && p.s.lit[0] == '0' && p.s.lit[1] == '0' {
-		p.illegal("expected number")
-		return nil
+	// Validate leading zeros: reject numbers like "01", "007", etc.
+	// Skip validation if prefix ends with '.' (like ".123")
+	hasDecimalPrefix := len(prefix) > 0 && prefix[len(prefix)-1] == '.'
+
+	if !hasDecimalPrefix && len(p.s.lit) > 1 && p.s.lit[0] == '0' {
+		// These are the only valid cases starting with '0':
+		isDecimal := p.s.lit[1] == '.'                                               // "0.123"
+		isScientific := len(p.s.lit) > 2 && (p.s.lit[1] == 'e' || p.s.lit[1] == 'E') // "0e5", "0E-3"
```

Comment:
These conditions where a little tough for my brain, are the new vars acceptable?